compiler/testData/diagnostics/tests/CompareToWithErrorType.fir.kt

```diff
@@ -1,6 +1,6 @@
 // !WITH_NEW_INFERENCE
 fun test() {
-  if (<!UNRESOLVED_REFERENCE!>x<!> <!UNRESOLVED_REFERENCE!>><!> 0) {
+  if (<!UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>x<!> > 0<!>) {
```

Comment:
Not sure this is out of scope, or something you can do while on it. But I feel like `visitErrorNamedReference` is a good place to do something regarding this.

So here, `>` is unresolved because `x` as an explicit receiver is unresolved. FE 1.0 reports kind of root cause, unresolved `x` only: https://github.com/JetBrains/kotlin/blob/master/compiler/testData/diagnostics/tests/CompareToWithErrorType.kt#L3

So, while visiting error named reference, if any of children elements has (another) error named reference, we could skip the outer-most one?

Reply:
OK, I will add another commit to this PR to skip reporting if the receiver has error.

Reply:
It's now added as the last commit.